src/main/java/org/opensrp/service/EventService.java

```diff
@@ -299,6 +299,13 @@ public void updateEvent(Event updatedEvent) {
 		updatedEvent.setDateEdited(DateTime.now());
 		updatedEvent.setServerVersion(allEvents.getNextServerVersion());
 		allEvents.update(updatedEvent);
+		String planIdentifier = updatedEvent.getDetails() != null ? updatedEvent.getDetails().get("planIdentifier") : null;
+		if (isPlanEvaluationEnabled && planIdentifier != null) {
+			PlanDefinition plan = planRepository.get(planIdentifier);
+			if (plan.getStatus().equals(PlanDefinition.PlanStatus.ACTIVE) && (plan.getEffectivePeriod().getEnd() == null
+					|| plan.getEffectivePeriod().getEnd().isAfter(LocalDate.now().toDateTimeAtStartOfDay())))
+				taskGenerator.processPlanEvaluation(plan, username, updatedEvent);
+		}
```

Comment:
Move this to a method so that it not repeated on addEvent. This also needs to be invoked on `addOrUpdateEvent()`

Reply:
Sure.